### PR TITLE
Add retries to testing script

### PR DESCRIPTION
**Pull request by polskikiel**
**Description**

Changes proposed in this pull request:

- Fixed Stability Checker logging

**Related issue(s)**

### CODE REVIEW

File: installation/scripts/testing-common.sh [hunk 1]
```diff
@@ -51,7 +69,8 @@ function printLogsFromFailedTests() {
         "Pending")
             log "'${pod}' failed due to too long Pending status" red
             printf "Fetching events from '${pod}':\n"
-            kubectl $(context_arg)  describe po ${pod} -n ${namespace} | awk 'x==1 {print} /Events:/ {x=1}'
+            executeKubectlWithRetries "kubectl $(context_arg)  describe po ${pod} -n ${namespace} | awk 'x==1 {print} /Events:/ {x=1}'"
+            echo $result
```
Comment (mszostok) [inline installation/scripts/testing-common.sh]: why print that?

File: installation/scripts/testing-common.sh [hunk 2]
```diff
@@ -7,24 +7,42 @@ function context_arg() {
...
+function executeKubectlWithRetries() {
+    local command="$1"
+    local retry=0
+
+    while [[ ${retry} -lt 5 ]]; do
+        result=$(${command})
+        if [[ $? -eq 0 ]]; then
+            return 0
+        else
+            echo ${result}
```
Comment (mszostok) [inline installation/scripts/testing-common.sh]: try to isolate that in dedicated shell, and return from this function only one output and get rid of the global `result` variable

File: installation/scripts/testing-common.sh [hunk 3]
```diff
@@ -86,9 +106,10 @@ function printLogsFromPod() {
     local namespace=$1 pod=$2
     local tailLimit=2000 bytesLimit=500000
     log "Fetching logs from '${pod}' with options tailLimit=${tailLimit} and bytesLimit=${bytesLimit}" nc bold
-    testPod=$(getContainerFromPod ${namespace} ${pod})
-    result=$(kubectl $(context_arg)  logs --tail=${tailLimit} --limit-bytes=${bytesLimit} -n ${namespace} -c ${testPod} ${pod})
-    if [ "${#result}" -eq 0 ]; then
+    container=$(getContainerFromPod ${namespace} ${pod})
+    echo $container
```
Comment (mszostok) [inline installation/scripts/testing-common.sh]: why print that?

Comment (polskikiel) [reply installation/scripts/testing-common.sh]: wasn't need

File: installation/scripts/testing-common.sh (hunk 1 above)
Comment (polskikiel) [reply installation/scripts/testing-common.sh]: i print that to log kubectl describe result

File: installation/scripts/testing-common.sh [hunk 4]
```diff
@@ -74,10 +94,12 @@ function getContainerFromPod() {
     local namespace="$1"
     local pod="$2"
     local containers2ignore="istio-init istio-proxy manager"
-    containersInPod=$(kubectl get pods ${pod} -o jsonpath='{.spec.containers[*].name}' -n ${namespace})
-    for container in $containersInPod; do
-        if [[ ! ${containers2ignore[*]} =~ "${container}" ]]; then
-            echo "${container}"
+
+    result=$(executeKubectlWithRetries "kubectl get pods ${pod} -o jsonpath={.spec.containers[*].name} -n ${namespace}")
```
Comment (mszostok) [inline installation/scripts/testing-common.sh]: how are u asserting the return code?

right now in err case, u will also iterate over err msg.

same problem for other executions

**Pull request merged**